src/App.jsx

```diff
@@ -14,33 +15,89 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [activeButton, setActiveButton] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const funcSortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveButton('alphabetically');
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const funcSortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (a.length === b.length) {
+        return goodsFromServer.indexOf(a) - goodsFromServer.indexOf(b);
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveButton('length');
+  };
+
+  const funcReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const funcReset = () => {
+    setGoods([...goodsFromServer]);
+    setActiveButton('');
+    setIsReversed(false);
+  };
+
+  const isOnStart = JSON.stringify(goodsFromServer) === JSON.stringify(goods);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${activeButton === 'alphabetically' ? '' : 'is-light'}`}
+          onClick={funcSortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${activeButton === 'length' ? '' : 'is-light'}`}
+          onClick={funcSortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+          onClick={funcReverse}
+        >
+          Reverse
+        </button>
+
+        {!isOnStart && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={funcReset}
+          >
+            Reset
+          </button>
+        )}
```

Comment:
The Reset button is conditionally rendered based on the `isOnStart` variable. This is a good practice to avoid unnecessary UI elements when they are not needed.